docs/architecture.md

```diff
@@ -0,0 +1,211 @@
+       ```
+     - Incorrect:
+        ```ts
+        // components
+        import { Avatar } from '../../../core/components/avatar';
+        import { TextField } from '../../../core/components/text-field';
+       
+        import type { BeautifulComponentProps } from 'modules/core/components/beautiful-component.interface.ts';       
+       
+        export const BeautifulComponent: FC<BeautifulComponentProps> = () => {}
+       ```
+   - All project imports should be grouped with comments (types, components, constants, etc...) except external dependencies and component interfaces and styles in scope of component folder (they should be divided with blank line)
+       - Correct:
+          ```ts
+          import React, { type FC, useEffect } from 'react';
+          // hooks
+          import { useSignInRequest } from '@/modules/auth/hooks/use-sign-in-request';
+          // components
+          import { Avatar, TextField } from '@/modules/core/components';
+         
+          import type { BeautifulComponentProps } from './beautiful-component.interface';
+          import { Root } from './beautiful-component.styled';
+         
+          export const BeautifulComponent: FC<BeautifulComponentProps> = () => {}
+         ```
+       - Incorrect:
+          ```ts
+          import React, { type FC, useEffect } from 'react';
+          import { Avatar, TextField } from '@/modules/core/components/avatar';
+          import { useSignInRequest } from '@/modules/auth/hooks/use-sign-in-request';
+          import type { BeautifulComponentProps } from './beautiful-component.interface';
+         
+          import { Root } from './beautiful-component.styled';
+         
+          export const BeautifulComponent: FC<BeautifulComponentProps> = () => {}
+         ```
+   - Use `type` prefix for all type imports
+     - Correct:
+        ```ts
+        import { type FC } from 'react';
+        import type { BeautifulComponentProps } from './beautiful-component.interface'
+         ```
+     - Incorrect:
+        ```ts
+        import { FC } from 'react';
+        import { BeautifulComponentProps } from './beautiful-component.interface';
+       ```
+> Note! If one of rules above must be violated for an important reason, please leave `todo` comment in the code and create an issue in the GitHub
+
+## <a name="folder-structure"></a> Folder structure
+
+1. **assets**
+
+    Static files, such as icons, images, fonts, etc... Each kind of asset should have an individual folder.
+    ```
+    src/
+      assets/
+        icons/
+          my-icon.svg
+        images/
+          my-image.jpg
+        fonts/
+   ```
+2. **declarations**
+
+    Global type declarations. 
+
+3. **modules**
+
+    Module - independent project entity that contains ui, data, domain logic of this entity. 
+    ```
+    src/
+      modules/
+        user/
+        location/
+        auth/
+   ```
+4. **modules/module-name/ui**
```

Comment:
In my opinion there is no need for the additional ui and data folders. Maybe we should discuss this.